From ba9551b766848d93e0df0da0e8ceea2744fe9462 Mon Sep 17 00:00:00 2001
From: Arne Morten Kvarving <spiff@kodi.tv>
Date: Wed, 5 Dec 2018 09:20:21 +0100
Subject: [PATCH] fixed: division by zero in addon installation

let totalsteps equal deps + 1 since we are also installing
the add-on itself.

fixes #14989
---
 xbmc/addons/AddonInstaller.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xbmc/addons/AddonInstaller.cpp b/xbmc/addons/AddonInstaller.cpp
index ea778c18dd7e..4077c10f4ce8 100644
--- a/xbmc/addons/AddonInstaller.cpp
+++ b/xbmc/addons/AddonInstaller.cpp
@@ -714,7 +714,7 @@ bool CAddonInstallJob::Install(const std::string &installFrom, const RepositoryP
   SetText(g_localizeStrings.Get(24079));
   auto deps = m_addon->GetDependencies();
 
-  unsigned int totalSteps = static_cast<unsigned int>(deps.size());
+  unsigned int totalSteps = static_cast<unsigned int>(deps.size()) + 1;
   if (ShouldCancel(0, totalSteps))
     return false;
 
@@ -786,7 +786,7 @@ bool CAddonInstallJob::Install(const std::string &installFrom, const RepositoryP
   }
 
   SetText(g_localizeStrings.Get(24086));
-  SetProgress(0);
+  SetProgress(static_cast<unsigned int>(100.0 * (totalSteps - 1.0) / totalSteps));
 
   CFilesystemInstaller fsInstaller;
   if (!fsInstaller.InstallToFilesystem(installFrom, m_addon->ID()))
